### ISSUE TITLE
[SETUP] configuration + Climate entity

### ISSUE THREAD

**Issue by felisida** [h4de5/home-assistant-vimar#3]
**Describe the problem**
I'm attaching the names of my components in By-me and the db
[names.pdf](https://github.com/h4de5/home-assistant-vimar/files/4744523/names.pdf)
[DB_Domotica Vimar 200507.db.zip](https://github.com/h4de5/home-assistant-vimar/files/4744526/DB_Domotica.Vimar.200507.db.zip)

**Check for some things first**
unfortunately I'm not a code expert

**Setup**
 - Environment: macos
 - Home-Assistant Version: [v0.110.5]

**Comment by h4de5** [comment]
Hello.
Can you have a look at your home-assistant.log (should be in the main configuration directory for home-assistant) for any error messages regarding to vimar_platform?

It seems you have renamed many of your devices - so the type assignement won't work. I'll try to get the correct type on some other way. thanks for the screenshot

**Comment by felisida** [comment]
the installer assigned names after installation, maybe someone changed it from the web server. if you need to manually change the name if you explain how to do it I'll try, in the meantime I attach all the configuration check with the errors
[CONFIGURATION check.pdf](https://github.com/h4de5/home-assistant-vimar/files/4744677/CONFIGURATION.check.pdf)

**Comment by felisida** [comment]
this is the error in main config
![Schermata 2020-06-08 alle 10 25 11](https://user-images.githubusercontent.com/51525150/84008685-75721c80-a972-11ea-9d5b-80cf39ed7491.jpg)
![Schermata 2020-06-08 alle 10 25 23](https://user-images.githubusercontent.com/51525150/84008693-7905a380-a972-11ea-8619-32c65aef82fd.jpg)

**Comment by h4de5** [comment]
Hi update to the latest version. 
if you downloaded it via git - you can simple make a "git pull" in that directory.
If you deployed using the zip downlod - remove the vimar_platform directory completely before extracting it there again.

**Comment by felisida** [comment]
thanks a lot, with this upgrade, I don't have more errors, but some warning that I will attach. unfortunately I can control only the shades, but not lights and switches
[warning on check.docx](https://github.com/h4de5/home-assistant-vimar/files/4745267/warning.on.check.docx)

**Comment by h4de5** [comment]
so you are saying you **can** control shades? therefore you are the first confirmed successful tester! Thats at least partly great news 🥳 Thank you for that!

The lights and switches are missing probably due to the different naming schema. 

is it possible, that you don't have single light or switch actors setup on the webserver, but instead only rooms or groups without a numerical ID? this could be a reason why they are not listed. as for the integration I don't know yet, how to handle such rooms or groups within home-assistant.

I will have a look at that in the evening today - maybe I can add something, to show all entities in home-assitant and let you remove those you don't need.

**Comment by felisida** [comment]
thanks to you for this great job and your assistance, I am truly grateful to you. I hope that you can solve the issue with light and switch components, and even in a short time the multimedia system, I'm attaching the historical and entities
![Schermata 2020-06-08 alle 13 17 46](https://user-images.githubusercontent.com/51525150/84025292-94c97380-a98b-11ea-834a-c219eeb01602.jpg)
![Schermata 2020-06-08 alle 13 16 28](https://user-images.githubusercontent.com/51525150/84025304-9a26be00-a98b-11ea-92e8-38111efadd9e.jpg)

**Comment by h4de5** [comment]
@felisida can you check if you have deactivated entities from the vimar platform in your home-assistant?

you can find them in the settings:

https://your-home-assistant-url/config/entities

and then activate all checkboxes in the search options like this:

![image](https://user-images.githubusercontent.com/6115324/84754499-c57b5f80-afc0-11ea-9027-ea4e6d54d050.png)

I found out, that I am missing some light as well - specially the groups that includes a whole level. I can see them beeing added to the home-assistant, but for some reason they are not active. maybe your lights are listed there as well.

**Comment by felisida** [comment]
Hi, in the pdf you will
[I made the check that you requested.pdf](https://github.com/h4de5/home-assistant-vimar/files/4786104/I.made.the.check.that.you.requested.pdf)
 find the tests that I have performed

**Comment by h4de5** [comment]
@felisida thank you that did help me. I just uploaded another fix.
The item "Cameretta" and "Studio" - are those groups like rooms? or Individual actors like lights?

best regards - thanks again for testing

**Comment by felisida** [comment]
thanks to you for your work, these two are room names, but in reality they are also the names of the light could it be a problem? with this update now I see lights and controlled sockets, but they do not work, and some of the arrows of the shades are disabled, but even those enabled do not work, attached you will found some screenshot 
[new test.pdf](https://github.com/h4de5/home-assistant-vimar/files/4786937/new.test.pdf)

**Comment by h4de5** [comment]
@felisida maybe this happends due to the same name of the room and the light. I updated the code again - so each entity get its own ID added to the name.
the errors in your list, do they happen during start up or only after you try to switch states?

in regards of the cover arrows: I tried to show the state of the cover within home-assistant. it may not work very well right now, because vimar does not give that information.

**Comment by felisida** [comment]
perfect, great job! now I see shades, lights and sockets (like light), even tough starting HA I have this warning
![Schermata 2020-06-16 alle 18 46 55](https://user-images.githubusercontent.com/51525150/84804550-3e031000-b003-11ea-958b-8da979d003f6.jpg)
about the cover arrows, it's a good idea, but i think in order to do a fine job it have to use the limit switches of shades, because the result if it is partially or totally down it the same
![Schermata 2020-06-16 alle 19 00 58](https://user-images.githubusercontent.com/51525150/84805479-9edf1800-b004-11ea-93ea-6af1ebca8832.jpg)
if you have the desire and time to program the climate and audio, I will be very happy to test the component, otherwise I can close the issue. thanks a lot lot lot

**Comment by h4de5** [comment]
I see. I will revert that change for the covers then.
About the other types: climate i want to integrate next. at least reading. Setting is another story. Audio would be nice but i dont have anything to test it with. Same with rgb dimmers or other device types. The problem is, that i would need to analyze the requests from and to webserver while using the Web GUI. Maybe if i find any of those off the market. Then i will be able to add it.

**Comment by h4de5** [comment]
btw. you should be able to click the greyed arrows in the covers as well. so even if it wrongly shows a partly closed cover as closed, you can still click the closing arrow.

**Comment by felisida** [comment]
unfortunately in my case the grey arrow are disabled, about the other components I can help you if i send you some info?

**Comment by alfienero** [comment]
Dear Andi I write you here because me and Felice shared a similar path with your integration! I tried the last component update and I confirm you that I can see and control all the lights now!!! (including the two bedsides that are configured as light but in reality are like smart plugs) 

Regarding the wishlist, beside climate control, for which I'm available if you need data/test, would be useful to have scenarios as well. I know that you can reproduce them directly in HA, but this could be a way to control alarm (I have an exit scenario that turns on alarm). Unfortunately I don't have RGB lights and audio devices. Ilevia software controls climate through setpoints and climate modes.

**Comment by h4de5** [comment]
@alfienero : the smart plugs - do they have different/additional settings that can be modifed through the WebUI? for all standard automation entities that do not follow the naming scheme , I fallback to lights - therefore they may show up in the wrong list. I am not yet sure, how I could enable customizing those entities. but ill leave that for later.

(reading) climate/thermostats should be in already (since yesterday) - Ill have a look at scenarios later today. 

for adding other stuff: Ill try to make a video of how I am "debugging" the webserver in order to get the correct calls for the integration. It's not that complicated, but maybe cumbersome if you are not used to web developer tools. The only other way would be to have access to an installation that does have those devices.

**Comment by alfienero** [comment]
In reality are not "smart plugs" it's just a switch that turn on/off what's connected in the plug (bedside lamp). Even in webserver is configured as a light and no other info are available neither in webserver and HA. 

Regarding climate I don't have new entities in vimar platform that measure temperatures, I've checked as you suggested even in disabled entitities. Maybe mine is a different model?

I'd be eager to support you in debugging my webserver if you teach me how to do or as an alternative to create you a temporary user to remotely access my webserver

**Comment by felisida** [comment]
ABSOLUTELY, I can use a tutorial made by yourself, or give you a remotely access, but I should use an old pc win because I work with my pc

**Comment by h4de5** [comment]
Thank you both for your support ;-)
I'll take some time for this later in the evening today.

**Comment by felisida** [comment]
ABOUT CLIMATE, did you post a new versione because I my ha there isn't.

**Comment by h4de5** [comment]
should be there. added it yesterday night. just fixed the idle state right now.

**Comment by felisida** [comment]
with this update now in the configuration check, I don't have more this warning
![Schermata 2020-06-16 alle 18 46 55](https://user-images.githubusercontent.com/51525150/84804550-3e031000-b003-11ea-958b-8da979d003f6.jpg)
but I don't see the climate value

**Comment by h4de5** [comment]
there should me much more messages from vimar - can lower the log level for vimar in your configuration.yaml:

```yaml
logger:
  default: warning
  logs:
    custom_components.vimar_platform: info
```

**Comment by felisida** [comment]
hi thi is the complete configuration check
2020-06-18 07:33:37 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-06-18 07:33:37 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for vimar_platform which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-06-18 07:33:37 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for smartthinq_sensors which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-06-18 07:33:37 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hpprinter which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-06-18 07:33:39 INFO (MainThread) [custom_components.vimar_platform.vimarlink] Vimar link initialized
2020-06-18 07:33:39 INFO (MainThread) [custom_components.vimar_platform] Vimar CA Certificate is already in place: rootCA.VIMAR.crt
2020-06-18 07:33:40 ERROR (MainThread) [homeassistant.components.yr.sensor] Retrying in 17 minutes: https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/1.9/?lat=44.63765843409103&lon=11.1798495054245&msl=21 returned 203
2020-06-18 07:33:40 ERROR (MainThread) [metno] https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/1.9/ returned 203
2020-06-18 07:33:40 ERROR (MainThread) [homeassistant.components.met.weather] Retrying in 20 minutes
2020-06-18 07:33:42 INFO (SyncWorker_6) [custom_components.vimar_platform.vimarlink] Vimar login successfull
2020-06-18 07:33:42 INFO (SyncWorker_9) [custom_components.vimar_platform.vimarlink] getDevices started
2020-06-18 07:33:42 INFO (SyncWorker_9) [custom_components.vimar_platform.vimarlink] getDevices ends - found 50 devices
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found unknown device: others/-/-
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform.cover] Vimar Cover started!
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Cameretta
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Studio
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform.cover] Vimar Cover complete!
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform.light] Vimar Light started!
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Blindata
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Cameretta
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Scale
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Studio
2020-06-18 07:33:42 INFO (MainThread) [custom_components.vimar_platform.light] Vimar Light complete!
2020-06-18 07:33:49 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up upnp platform for sensor
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/homeassistant/helpers/entity_platform.py", line 178, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/tasks.py", line 483, in wait_for
    return fut.result()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/homeassistant/components/upnp/sensor.py", line 89, in async_setup_entry
    device: Device = hass.data[DOMAIN]["devices"][udn]
KeyError: 'uuid:3ce75827-487e-486b-8f59-de39b048b762'
2020-06-18 07:34:00 ERROR (MainThread) [metno] https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/1.9/ returned 
2020-06-18 07:34:00 ERROR (MainThread) [homeassistant.components.met.weather] Retrying in 17 minutes
2020-06-18 07:34:00 ERROR (MainThread) [metno] https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/1.9/ returned 203
2020-06-18 07:34:00 ERROR (MainThread) [homeassistant.components.met.weather] Retrying in 20 minutes
2020-06-18 07:34:08 ERROR (MainThread) [homeassistant.config_entries] Error unloading entry OnHub for upnp
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/homeassistant/config_entries.py", line 303, in async_unload
    result = await component.async_unload_entry(  # type: ignore
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/homeassistant/components/upnp/__init__.py", line 144, in async_unload_entry
    del hass.data[DOMAIN]["devices"][udn]
KeyError: 'uuid:3ce75827-487e-486b-8f59-de39b048b762'

**Comment by felisida** [comment]
good morning, before when I started HA, i found this situation for my cover
![Schermata 2020-06-18 alle 08 03 53](https://user-images.githubusercontent.com/51525150/84984252-c2a47a00-b13a-11ea-95e9-fc94829e63c5.jpg)
but they was all closed, however after started my routine to go up the covers the status come back aligned

**Comment by h4de5** [comment]
I only get the last direction from the cover and if it has just been stopped. Also as far as i can see, i can not say in HA that something is partly open.

You can try again the climates.

**Comment by alfienero** [comment]
Good morning Andi, on my side now I can see two new climate entitities even if I'm not able to control them but maybe just because I don't know what I have to do! I can't even read the measured temperatures (null).

**Comment by h4de5** [comment]
looks like we have different devices with different status fields then .. CH_Clima and CH_HVAC_NoZonaNeutra

**Comment by alfienero** [comment]
I have this attributes for both:
hvac_modes: heat, cool, off
min_temp: 7
max_temp: 35
current_temperature: null
temperature: null
hvac_action: off
friendly_name: Termostato camera
icon: mdi:home-thermometer-outline
supported_features: 1

**Comment by felisida** [comment]
hi Andi, sorry for the delay in my reply, but I don't receive more the notifications...
I installed the last versione and now I see this one, don't show the temperature is it normal?
![Schermata 2020-06-19 alle 17 07 44](https://user-images.githubusercontent.com/51525150/85148565-9549f000-b250-11ea-97f5-396643b63ddd.jpg)

**Comment by felisida** [comment]
this is the configuration check:
2020-06-19 17:02:35 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-06-19 17:02:35 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hpprinter which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-06-19 17:02:35 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for smartthinq_sensors which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-06-19 17:02:35 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for vimar_platform which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-06-19 17:02:36 INFO (MainThread) [custom_components.vimar_platform.vimarlink] Vimar link initialized
2020-06-19 17:02:36 INFO (MainThread) [custom_components.vimar_platform] Vimar CA Certificate is already in place: rootCA.VIMAR.crt
2020-06-19 17:02:36 WARNING (MainThread) [homeassistant.components.mobile_app.notify] Found duplicate device name iPhone di Felice
2020-06-19 17:02:36 WARNING (MainThread) [homeassistant.components.mobile_app.notify] Found duplicate device name iPhone di Felice
2020-06-19 17:02:38 INFO (SyncWorker_6) [custom_components.vimar_platform.vimarlink] Vimar login successfull
2020-06-19 17:02:39 INFO (SyncWorker_3) [custom_components.vimar_platform.vimarlink] getDevices started
2020-06-19 17:02:39 INFO (SyncWorker_3) [custom_components.vimar_platform.vimarlink] getDevices ends - found 50 devices
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Unsupported object returned from web server: CH_Audio / Bagno Camera
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Unsupported object returned from web server: CH_Audio / Bagno ospiti
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Unsupported object returned from web server: CH_Audio / Camera matrimoniale
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Unsupported object returned from web server: CH_Audio / Camera ospiti
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Unsupported object returned from web server: CH_Audio / Cameretta
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Unsupported object returned from web server: CH_Audio / Cucina
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Unsupported object returned from web server: CH_Audio / Soggiorno
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Unsupported object returned from web server: CH_Audio / Studio
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform.climate] Vimar Climate started!
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Cameretta
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Cucina
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Soggiorno
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Studio
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform.climate] Vimar Climate complete!
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform.cover] Vimar Cover started!
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Cameretta
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Studio
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform.cover] Vimar Cover complete!
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform.light] Vimar Light started!
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Blindata
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Cameretta
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Scale
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform] Found a device with an uncommon naming schema: Studio
2020-06-19 17:02:39 INFO (MainThread) [custom_components.vimar_platform.light] Vimar Light complete!

**Comment by h4de5** [comment]
@felisida @alfienero 

I have setup a small video of how I read out data for new devices - check out if you can follow the descriptions:

https://github.com/h4de5/home-assistant-vimar/blob/master/Contribute.md

best regards!

**Comment by felisida** [comment]
this can help you?
<?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:ns="urn:xmethods-dpadws">
    <SOAP-ENV:Body SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
        <ns:service-databasesocketoperationResponse>
            <result-len>Response payload length: 5792</result-len>
            <result>DPCM-0000</result>
            <payload-len>5792</payload-len>
            <payload>Response: DBMG-000&#xA;NextRows: 119&#xA;Row000001: 'ID','NAME','STATUS_ID','CURRENT_VALUE'&#xA;Row000002: '229','_DPAD_DBCONSTANT_VIMAR_BYME_TRIGGEROBJECT_VIDEOMESSAGE_UNHANDLED','-1','0'&#xA;Row000003: '265','_DPAD_DBCONSTANT_VIMAR_BYME_TRIGGEROBJECT_SENSORS_HUMIDITY_ALARM','-1','0'&#xA;Row000004: '270','_DPAD_DBCONSTANT_VIMAR_BYME_TRIGGEROBJECT_SAI2_LOG_ALARM','-1','1'&#xA;Row000005: '794','forzatura off','-1','0'&#xA;Row000006: '796','allarme_massetto','-1','0'&#xA;Row000007: '798','regolazione','-1','1'&#xA;Row000008: '802','modalita_fancoil','-1','0'&#xA;Row000009: '804','velocita_fancoil','-1','0'&#xA;Row000010: '806','funzionamento','-1','1'&#xA;Row000011: '814','setpoint','-1','27.00'&#xA;Row000012: '822','temporizzazione','-1','1'&#xA;Row000013: '833','temperatura_misurata','-1','27.20'&#xA;Row000014: '836','stato_boost on/off','-1','0'&#xA;Row000015: '837','stato_principale_condizionamento on/off','-1','1'&#xA;Row000016: '838','stato_principale_riscaldamento on/off','-1','0'&#xA;Row000017: '848','forzatura off','-1','0'&#xA;Row000018: '850','allarme_massetto','-1','0'&#xA;Row000019: '852','regolazione','-1','1'&#xA;Row000020: '856','modalita_fancoil','-1','0'&#xA;Row000021: '858','velocita_fancoil','-1','0'&#xA;Row000022: '860','funzionamento','-1','1'&#xA;Row000023: '868','setpoint','-1','27.00'&#xA;Row000024: '876','temporizzazione','-1','1'&#xA;Row000025: '887','temperatura_misurata','-1','26.70'&#xA;Row000026: '890','stato_boost on/off','-1','0'&#xA;Row000027: '891','stato_principale_condizionamento on/off','-1','0'&#xA;Row000028: '892','stato_principale_riscaldamento on/off','-1','0'&#xA;Row000029: '902','forzatura off','-1','0'&#xA;Row000030: '904','allarme_massetto','-1','0'&#xA;Row000031: '906','regolazione','-1','1'&#xA;Row000032: '910','modalita_fancoil','-1','0'&#xA;Row000033: '912','velocita_fancoil','-1','0'&#xA;Row000034: '914','funzionamento','-1','1'&#xA;Row000035: '922','setpoint','-1','27.60'&#xA;Row000036: '930','temporizzazione','-1','1'&#xA;Row000037: '941','temperatura_misurata','-1','27.20'&#xA;Row000038: '944','stato_boost on/off','-1','0'&#xA;Row000039: '945','stato_principale_condizionamento on/off','-1','0'&#xA;Row000040: '946','stato_principale_riscaldamento on/off','-1','0'&#xA;Row000041: '956','forzatura off','-1','0'&#xA;Row000042: '958','allarme_massetto','-1','0'&#xA;Row000043: '960','regolazione','-1','1'&#xA;Row000044: '964','modalita_fancoil','-1','0'&#xA;Row000045: '966','velocita_fancoil','-1','0'&#xA;Row000046: '968','funzionamento','-1','1'&#xA;Row000047: '976','setpoint','-1','27.00'&#xA;Row000048: '984','temporizzazione','-1','1'&#xA;Row000049: '995','temperatura_misurata','-1','26.90'&#xA;Row000050: '998','stato_boost on/off','-1','0'&#xA;Row000051: '999','stato_principale_condizionamento on/off','-1','0'&#xA;Row000052: '1000','stato_principale_riscaldamento on/off','-1','0'&#xA;Row000053: '1010','forzatura off','-1','0'&#xA;Row000054: '1012','allarme_massetto','-1','0'&#xA;Row000055: '1014','regolazione','-1','1'&#xA;Row000056: '1018','modalita_fancoil','-1','0'&#xA;Row000057: '1020','velocita_fancoil','-1','0'&#xA;Row000058: '1022','funzionamento','-1','1'&#xA;Row000059: '1030','setpoint','-1','27.00'&#xA;Row000060: '1038','temporizzazione','-1','1'&#xA;Row000061: '1049','temperatura_misurata','-1','25.20'&#xA;Row000062: '1052','stato_boost on/off','-1','0'&#xA;Row000063: '1053','stato_principale_condizionamento on/off','-1','0'&#xA;Row000064: '1054','stato_principale_riscaldamento on/off','-1','0'&#xA;Row000065: '1064','forzatura off','-1','0'&#xA;Row000066: '1066','allarme_massetto','-1','0'&#xA;Row000067: '1068','regolazione','-1','1'&#xA;Row000068: '1072','modalita_fancoil','-1','0'&#xA;Row000069: '1074','velocita_fancoil','-1','0'&#xA;Row000070: '1076','funzionamento','-1','1'&#xA;Row000071: '1084','setpoint','-1','27.00'&#xA;Row000072: '1092','temporizzazione','-1','1'&#xA;Row000073: '1103','temperatura_misurata','-1','25.60'&#xA;Row000074: '1106','stato_boost on/off','-1','0'&#xA;Row000075: '1107','stato_principale_condizionamento on/off','-1','0'&#xA;Row000076: '1108','stato_principale_riscaldamento on/off','-1','0'&#xA;Row000077: '1876','uscita4','-1','non_utilizzata'&#xA;Row000078: '1877','uscita3','-1','non_utilizzata'&#xA;Row000079: '1878','uscita2','-1','non_utilizzata'&#xA;Row000080: '1879','uscita1','-1','CH_Uscita_ValvolaOnOff'&#xA;Row000081: '1935','uscita4','-1','non_utilizzata'&#xA;Row000082: '1936','uscita3','-1','non_utilizzata'&#xA;Row000083: '1937','uscita2','-1','non_utilizzata'&#xA;Row000084: '1938','uscita1','-1','CH_Uscita_ValvolaOnOff'&#xA;Row000085: '1994','uscita4','-1','non_utilizzata'&#xA;Row000086: '1995','uscita3','-1','non_utilizzata'&#xA;Row000087: '1996','uscita2','-1','non_utilizzata'&#xA;Row000088: '1997','uscita1','-1','CH_Uscita_ValvolaOnOff'&#xA;Row000089: '2053','uscita4','-1','non_utilizzata'&#xA;Row000090: '2054','uscita3','-1','non_utilizzata'&#xA;Row000091: '2055','uscita2','-1','non_utilizzata'&#xA;Row000092: '2056','uscita1','-1','CH_Uscita_ValvolaOnOff'&#xA;Row000093: '2112','uscita4','-1','non_utilizzata'&#xA;Row000094: '2113','uscita3','-1','non_utilizzata'&#xA;Row000095: '2114','uscita2','-1','non_utilizzata'&#xA;Row000096: '2115','uscita1','-1','CH_Uscita_ValvolaOnOff'&#xA;Row000097: '2171','uscita4','-1','non_utilizzata'&#xA;Row000098: '2172','uscita3','-1','non_utilizzata'&#xA;Row000099: '2173','uscita2','-1','non_utilizzata'&#xA;Row000100: '2174','uscita1','-1','CH_Uscita_ValvolaOnOff'&#xA;Row000101: '8548','_DPAD_PRODUCT_VIMARBYME_CERTIFICATE_TRIGGER','-1','0'&#xA;Row000102: '9763','T1','-1','20.0'&#xA;Row000103: '9764','T2','-1','22.0'&#xA;Row000104: '9765','T3','-1','24'&#xA;Row000105: '9818','T1','-1','20.0'&#xA;Row000106: '9819','T2','-1','22.0'&#xA;Row000107: '9820','T3','-1','24'&#xA;Row000108: '9873','T1','-1','20.0'&#xA;Row000109: '9874','T2','-1','22.0'&#xA;Row000110: '9875','T3','-1','24'&#xA;Row000111: '9928','T1','-1','20.0'&#xA;Row000112: '9929','T2','-1','22.0'&#xA;Row000113: '9930','T3','-1','24.0'&#xA;Row000114: '9983','T1','-1','20.0'&#xA;Row000115: '9984','T2','-1','22.0'&#xA;Row000116: '9985','T3','-1','24'&#xA;Row000117: '10038','T1','-1','20.0'&#xA;Row000118: '10039','T2','-1','22.0'&#xA;Row000119: '10040','T3','-1','24'&#xA;</payload>
        </ns:service-databasesocketoperationResponse>
    </SOAP-ENV:Body>
</SOAP-ENV:Envelope>

**Comment by felisida** [comment]
hi with the last update I can see the set points and I can act on the cursor, but I don't see the current temperature
![Schermata 2020-06-22 alle 09 18 59](https://user-images.githubusercontent.com/51525150/85259823-ff89ad00-b469-11ea-8624-0f79646ef66d.jpg)

**Comment by alfienero** [comment]
Hi andi, sorry for the late reply, I did the first step let me know if you need attributes as well. The below code is extracted from functions-climate with 2 thermostats as devices. Starting from row n.60 I can see something related to those devices. 

Row 60: could be turn on / off
Row 61-63: honestly I don't know
Row 64-65 mode and speed for cooling
Row000064: '2838','modalita_fancoil','-1','0'
Row000065: '2840','velocita_fancoil'

Row 66 I guess "funzionamento" is the climate mode of work from 0 to 3: 
0 Automatico -> Automatic (follow the 3-setpoints T1-T2-T3 program along the week see rows 97-99)
1 Manuale-> Manual mode with setpoint
2 Riduzione -> I guess a kind of energy saving mode, never used
3 Assenza -> Away mode (If you use an extreme setpoint - eg. 31° for cooling - is equivalent of being off)

Row 67: single setpoint, valid for all climate modes except automatic

Row 69 is the value that the thermometer reads now
Row000069: '2869','temperatura_misurata','-1','27.20' 
`
<payload
>
Response: DBMG-000
NextRows: 99
Row000001: 'ID','NAME','STATUS_ID','CURRENT_VALUE'
Row000002: '198','_DPAD_DBCONSTANT_VIMAR_BYME_TRIGGEROBJECT_SAI_LOG_ALARM','-1','0'
Row000003: '229','_DPAD_DBCONSTANT_VIMAR_BYME_TRIGGEROBJECT_VIDEOMESSAGE_UNHANDLED','-1','0'
Row000004: '265','_DPAD_DBCONSTANT_VIMAR_BYME_TRIGGEROBJECT_SENSORS_HUMIDITY_ALARM','-1','0'
Row000005: '363','control_dimming','-1','1'
Row000006: '365','direction_dimming','-1','1'
Row000007: '367','on/off','-1','0'
Row000008: '369','value','-1','38'
Row000009: '375','control_dimming','-1','0'
Row000010: '377','direction_dimming','-1','0'
Row000011: '379','on/off','-1','0'
Row000012: '381','value','-1','0'
Row000013: '387','control_dimming','-1','0'
Row000014: '389','direction_dimming','-1','0'
Row000015: '391','on/off','-1','0'
Row000016: '395','control_dimming','-1','0'
Row000017: '397','direction_dimming','-1','0'
Row000018: '399','on/off','-1','0'
Row000019: '401','value','-1','0'
Row000020: '407','control_dimming','-1','0'
Row000021: '409','direction_dimming','-1','0'
Row000022: '411','on/off','-1','0'
Row000023: '413','value','-1','0'
Row000024: '419','control_dimming','-1','0'
Row000025: '421','direction_dimming','-1','0'
Row000026: '423','on/off','-1','0'
Row000027: '427','control_dimming','-1','0'
Row000028: '429','direction_dimming','-1','0'
Row000029: '431','on/off','-1','0'
Row000030: '435','control_dimming','-1','0'
Row000031: '437','direction_dimming','-1','0'
Row000032: '439','on/off','-1','0'
Row000033: '443','control_dimming','-1','0'
Row000034: '445','direction_dimming','-1','0'
Row000035: '447','on/off','-1','0'
Row000036: '451','control_dimming','-1','0'
Row000037: '453','direction_dimming','-1','0'
Row000038: '455','on/off','-1','0'
Row000039: '459','control_dimming','-1','0'
Row000040: '461','direction_dimming','-1','1'
Row000041: '463','on/off','-1','0'
Row000042: '465','value','-1','50'
Row000043: '471','control_dimming','-1','0'
Row000044: '473','direction_dimming','-1','0'
Row000045: '475','on/off','-1','0'
Row000046: '479','control_dimming','-1','0'
Row000047: '481','direction_dimming','-1','0'
Row000048: '483','on/off','-1','0'
Row000049: '487','control_dimming','-1','0'
Row000050: '489','direction_dimming','-1','0'
Row000051: '491','on/off','-1','0'
Row000052: '495','control_dimming','-1','0'
Row000053: '497','direction_dimming','-1','0'
Row000054: '499','on/off','-1','0'
Row000055: '504','control_dimming','-1','0'
Row000056: '506','direction_dimming','-1','0'
Row000057: '508','on/off','-1','0'
Row000058: '2376','control_dimming','-1','0'
Row000059: '2378','direction_dimming','-1','0'
Row000060: '2380','on/off','-1','0'
Row000061: '2830','forzatura off','-1','0'
Row000062: '2832','allarme_massetto','-1','0'
Row000063: '2834','regolazione','-1','1'
Row000064: '2838','modalita_fancoil','-1','0'
Row000065: '2840','velocita_fancoil','-1','0'
Row000066: '2842','funzionamento','-1','3'
Row000067: '2850','setpoint','-1','31.00'
Row000068: '2858','temporizzazione','-1','1'
Row000069: '2869','temperatura_misurata','-1','27.20'
Row000070: '2872','stato_boost on/off','-1','0'
Row000071: '2873','stato_principale_condizionamento on/off','-1','0'
Row000072: '2874','stato_principale_riscaldamento on/off','-1','0'
Row000073: '2888','forzatura off','-1','0'
Row000074: '2890','allarme_massetto','-1','0'
Row000075: '2892','regolazione','-1','1'
Row000076: '2896','modalita_fancoil','-1','0'
Row000077: '2898','velocita_fancoil','-1','0'
Row000078: '2900','funzionamento','-1','3'
Row000079: '2908','setpoint','-1','31.00'
Row000080: '2916','temporizzazione','-1','1'
Row000081: '2927','temperatura_misurata','-1','28.40'
Row000082: '2930','stato_boost on/off','-1','0'
Row000083: '2931','stato_principale_condizionamento on/off','-1','0'
Row000084: '2932','stato_principale_riscaldamento on/off','-1','0'
Row000085: '3148','uscita4','-1','non_utilizzata'
Row000086: '3149','uscita3','-1','non_utilizzata'
Row000087: '3150','uscita2','-1','CH_Uscita_FancoilProporzionale'
Row000088: '3151','uscita1','-1','CH_Uscita_ValvolaOnOff'
Row000089: '3219','uscita4','-1','CH_Uscita_ValvolaOnOff'
Row000090: '3220','uscita3','-1','CH_Uscita_ValvolaOnOff'
Row000091: '3221','uscita2','-1','CH_Uscita_FancoilProporzionale'
Row000092: '3222','uscita1','-1','CH_Uscita_ValvolaOnOff'
Row000093: '5041','_DPAD_PRODUCT_VIMARBYME_CERTIFICATE_TRIGGER','-1','0'
Row000094: '5361','T1','-1',''
Row000095: '5362','T2','-1',''
Row000096: '5363','T3','-1',''
Row000097: '5430','T1','-1','27.4'
Row000098: '5431','T2','-1','28.0'
Row000099: '5432','T3','-1','29.0'`

**Comment by felisida** [comment]
this is the code for the audio:
<?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:ns="urn:xmethods-dpadws">
    <SOAP-ENV:Body SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
        <ns:service-databasesocketoperationResponse>
            <result-len>Response payload length: 1589</result-len>
            <result>DPCM-0000</result>
            <payload-len>1589</payload-len>
            <payload>Response: DBMG-000&#xA;NextRows: 37&#xA;Row000001: 'ID','NAME','STATUS_ID','CURRENT_VALUE'&#xA;Row000002: '229','_DPAD_DBCONSTANT_VIMAR_BYME_TRIGGEROBJECT_VIDEOMESSAGE_UNHANDLED','-1','0'&#xA;Row000003: '265','_DPAD_DBCONSTANT_VIMAR_BYME_TRIGGEROBJECT_SENSORS_HUMIDITY_ALARM','-1','0'&#xA;Row000004: '270','_DPAD_DBCONSTANT_VIMAR_BYME_TRIGGEROBJECT_SAI2_LOG_ALARM','-1','1'&#xA;Row000005: '1352','channel','-1','4'&#xA;Row000006: '1368','on/off','-1','0'&#xA;Row000007: '1376','volume','-1','0'&#xA;Row000008: '1377','source','-1','4'&#xA;Row000009: '1398','channel','-1','4'&#xA;Row000010: '1414','on/off','-1','0'&#xA;Row000011: '1422','volume','-1','0'&#xA;Row000012: '1423','source','-1','4'&#xA;Row000013: '1428','channel','-1','4'&#xA;Row000014: '1444','on/off','-1','0'&#xA;Row000015: '1452','volume','-1','7'&#xA;Row000016: '1453','source','-1','4'&#xA;Row000017: '1458','channel','-1','4'&#xA;Row000018: '1474','on/off','-1','0'&#xA;Row000019: '1482','volume','-1','0'&#xA;Row000020: '1483','source','-1','4'&#xA;Row000021: '1488','channel','-1','4'&#xA;Row000022: '1504','on/off','-1','0'&#xA;Row000023: '1512','volume','-1','0'&#xA;Row000024: '1513','source','-1','4'&#xA;Row000025: '2505','channel','-1','4'&#xA;Row000026: '2522','on/off','-1','0'&#xA;Row000027: '2530','volume','-1','80'&#xA;Row000028: '2531','source','-1','4'&#xA;Row000029: '2536','channel','-1','0'&#xA;Row000030: '2553','on/off','-1','0'&#xA;Row000031: '2561','volume','-1','46'&#xA;Row000032: '2562','source','-1','4'&#xA;Row000033: '2567','channel','-1','4'&#xA;Row000034: '2583','on/off','-1','0'&#xA;Row000035: '2591','volume','-1','0'&#xA;Row000036: '2592','source','-1','4'&#xA;Row000037: '8548','_DPAD_PRODUCT_VIMARBYME_CERTIFICATE_TRIGGER','-1','0'&#xA;</payload>
        </ns:service-databasesocketoperationResponse>
    </SOAP-ENV:Body>
</SOAP-ENV:Envelope>

**Comment by felisida** [comment]
<?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:ns="urn:xmethods-dpadws">
    <SOAP-ENV:Body SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
        <ns:service-databasesocketoperationResponse>
            <result-len>Response payload length: 721</result-len>
            <result>DPCM-0000</result>
            <payload-len>721</payload-len>
            <payload>Response: DBMG-000&#xA;NextRows: 11&#xA;Row000001: 'ID','NAME','STATUS_ID','CURRENT_VALUE'&#xA;Row000002: '229','_DPAD_DBCONSTANT_VIMAR_BYME_TRIGGEROBJECT_VIDEOMESSAGE_UNHANDLED','-1','0'&#xA;Row000003: '265','_DPAD_DBCONSTANT_VIMAR_BYME_TRIGGEROBJECT_SENSORS_HUMIDITY_ALARM','-1','0'&#xA;Row000004: '270','_DPAD_DBCONSTANT_VIMAR_BYME_TRIGGEROBJECT_SAI2_LOG_ALARM','-1','1'&#xA;Row000005: '286','scambio_totale','-1','-2.510'&#xA;Row000006: '288','consumo_totale','-1','0.470'&#xA;Row000007: '290','produzione_totale','-1','2.980'&#xA;Row000008: '292','immissione_totale','-1','2.510'&#xA;Row000009: '294','prelievo_totale','-1','0.000'&#xA;Row000010: '296','autoconsumo_totale','-1','0.470'&#xA;Row000011: '8548','_DPAD_PRODUCT_VIMARBYME_CERTIFICATE_TRIGGER','-1','0'&#xA;</payload>
        </ns:service-databasesocketoperationResponse>
    </SOAP-ENV:Body>
</SOAP-ENV:Envelope>

**Comment by felisida** [comment]
if you need of any stuff you have only to told me

**Comment by alfienero** [comment]
> hi with the last update I can see the set points and I can act on the cursor, but I don't see the current temperature
> ![Schermata 2020-06-22 alle 09 18 59](https://user-images.githubusercontent.com/51525150/85259823-ff89ad00-b469-11ea-8624-0f79646ef66d.jpg)

Same for me I can set setpoints but can't see the instant measured temperature ('temperatura_misurata').

Regarding scenarios, all of them are charged as switch and seems to works properly!!!! Thank you so much again!

**Comment by h4de5** [comment]
@felisida @alfienero - tried to add alternative temperature attribute.

in regards of the `funzionamento` attribute: it seems there is a difference between our two devices. mine is just a thermostat (art.Nr. 14514) it has status:
* 0 (idle)
* 6 (manual setting)
* 7 (manual for a certain time)
* 8 (auto according to plan)

I will add the new 4 modes as you described above and try what I can do with the fancoil.

Please note that it will only be possible to set temperatures for "manual" mods and, by the click onto the automatic-Mode button, fallback to the temperatures defiend previously in T1 to T3. The reason for is that, I do not get easily the values for T1 to T3 and afaik Hass does not support settings multible target values.

**Comment by felisida** [comment]
hi Andi, there is a issue with new upgrade;
![Schermata 2020-06-22 alle 12 48 18](https://user-images.githubusercontent.com/51525150/85279550-e3e0cf80-b486-11ea-8b56-4cc1484bafa9.jpg)
2020-06-22 12:47:41 ERROR (SyncWorker_6) [custom_components.vimar_platform.vimarlink] Error parsing XML: TypeError("a bytes-like object is required, not 'bool'")
2020-06-22 12:47:41 ERROR (SyncWorker_6) [custom_components.vimar_platform.vimarlink] Problematic XML: False
2020-06-22 12:47:41 ERROR (MainThread) [homeassistant.setup] Error during setup of component vimar_platform
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/homeassistant/setup.py", line 190, in _async_setup_component
    result = await asyncio.wait_for(task, SLOW_SETUP_MAX_WAIT)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/tasks.py", line 483, in wait_for
    return fut.result()
  File "/Users/felice/.homeassistant/custom_components/vimar_platform/__init__.py", line 99, in async_setup
    valid_login = await hass.async_add_executor_job(vimarconnection.check_login)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/felice/.homeassistant/custom_components/vimar_platform/vimarlink.py", line 131, in check_login
    self.login()
  File "/Users/felice/.homeassistant/custom_components/vimar_platform/vimarlink.py", line 108, in login
    logincode = xml.find('result')
AttributeError: 'NoneType' object has no attribute 'find'

**Comment by alfienero** [comment]
> thermostat (art.Nr. 14514

I checked better the modes and I found out this:
0 (automatic)
1 (manual)
2 (so called 'reduction')
3 (away)
5 (manual for a certain time)
6 (off)

**Comment by h4de5** [comment]
@felisida : can you check if the web server is still running and responding to your normal web requests? it may need a cold restart if not.

**Comment by felisida** [comment]
you right! I hade to restar the web server 
![Schermata 2020-06-22 alle 13 05 18](https://user-images.githubusercontent.com/51525150/85280793-21465c80-b489-11ea-8cf1-2b596ccdf821.jpg)

**Comment by h4de5** [comment]
great. current mode seems not yet right - will add it to the next updates. thanks again to all of you, for keeping up with the tests :D

**Comment by alfienero** [comment]
I confirm that now I can read the measured temperature!! Thanks again for your impressive work!

**Comment by h4de5** [comment]
@felisida @alfienero 
can you enable the fancoil or give me a range of possible values for `modalita_fancoil` and `velocita_fancoil` ? it would be enough to enable it, turn it half way and full way up and check for the response like above.

**Comment by alfienero** [comment]
Hi Andi I can’t do it this week since I’m out. As soon as I get back home I’ll do it!

**Comment by h4de5** [comment]
just for an update - I tried to add proper support for the individual modes (on, off, auto, manual) in regards of our two different climate device types. I too added support for fancoil settings in the climate module.

**Comment by felisida** [comment]
I'm sorry but my system don't have fan-coil, mine is a floor heating with heat pump

**Comment by h4de5** [comment]
@felisida : I now tried to add external heating sources as well. Can you send me a screenshot of how this looks like on your home-assistant screen (just small screenshot for the docs).

**Comment by felisida** [comment]
![Schermata 2020-06-29 alle 10 53 09](https://user-images.githubusercontent.com/51525150/85993552-cd8ec280-b9f6-11ea-9d5c-1eab74de9c1c.jpg)
this one?

**Comment by h4de5** [comment]
yes. I hoped the heat pump option would show up as an extra button. what comes up if you select the 3 dots on top?

**Comment by felisida** [comment]
I can't reach to sent you a screenshot

**Comment by h4de5** [comment]
ah ok. but at least the button is there :D
as far as I can see, this ticket can also be considered done, leaving only alarm and audio integration for the next goals.

I tried some things with the media player yesterday but may need some more time on that, because I can't test it properly. therefore I think I will first fix some issues on the home-assistant part and maybe add [HACS support](https://github.com/h4de5/home-assistant-vimar/issues/10) first.

**Comment by felisida** [comment]
hi handy I think there is a little bug in the climate project, this morning I tried to start the heating system, and all' work properly but I think there is a little issue In the translate because the correct description is "Riscaldamento" e non "Raffreddamento"
![Schermata 2020-09-28 alle 11 23 29](https://user-images.githubusercontent.com/51525150/94415477-df127280-017d-11eb-8594-d5aefd7f0e43.jpg)

**Comment by felisida** [comment]
too in this condition the command was send properly but the correct label is "Raffreddamento"
![Schermata 2020-09-28 alle 11 43 26](https://user-images.githubusercontent.com/51525150/94417090-f05c7e80-017f-11eb-8ac2-3ef69a25f72a.jpg)

**Comment by h4de5** [comment]
@felisida the translation comes directly from the home-assistant project. I only change the states.
I recently changed around some of the states, just to find out, that there seems to exist at least 3 different climate devices from vimar (see converstation in home-assistant forum). maybe the states are still not right.
I uploaded a new debug version for cliemates - if you set one thermostat to heating, one to cooling and one off, you should be able to see the startup states for each thermostate in the home-assistant log file.

If you say, that your vimar thermostate indeed changes to the requested states - (switches to cooling if pressed on cooling etc.) only than it must really a translation error - but that is very unlikely as it would affect all italian hass installations.

**Comment by felisida** [comment]
it's strange! all work properly only the label is incorrect, but this doesn't affect the operation

**Comment by felisida** [comment]
I UPDATET THE last repository, but the pairing is still incorrect

**Comment by h4de5** [comment]
just checked the global language settings - as far as I can see, the translation seems fine:
https://github.com/home-assistant/frontend/blob/da0658f3d99109170137251e34b03bf21a6af478/translations/frontend/it.json#L34

very strange :\
can you try to set the target temperatur above your current temperature to see if it says "Riscaldamento"?

**Comment by felisida** [comment]
the translation is correct,
but the association in HA the behavior is reverse
![Schermata 2020-09-28 alle 15 20 24](https://user-images.githubusercontent.com/51525150/94437617-4b04d300-019e-11eb-9691-58714a2820ec.jpg)
another fact, in cooling state if I make change in HA the label is always "inattivo" like as can't reach to read the state from web server, but only this one, on the contrary when I made change directly from web server I had with you see in the screenshot

**Comment by h4de5** [comment]
@felisida please allow some time between changing the states in HA and getting the actual state from the webserver.
it is possible, that you switch to cooling, in one cycle, and need to wait two more cycles (around 8 seconds each) until HASS has updated its setting in the frontend.
i will have a look again later today - maybe I find another issue. the problem again, I am afraid I can only take guesses whats wrong.

the states has been changed due to a bug report from the community - https://community.home-assistant.io/t/vimar-integration-wip/202073/27

**Comment by felisida** [comment]
hello i imagined time problems but i waited at least 1 minute and also did a page refresh to force but the status does not change

**Comment by h4de5** [comment]
@felisida I think I fixed the problem with the translation now.

**Comment by felisida** [comment]
Perfect work fine!!!
![Schermata 2020-10-20 alle 13 36 26](https://user-images.githubusercontent.com/51525150/96580888-4339d980-12d9-11eb-8573-29e76905c72d.jpg)

**Comment by h4de5** [comment]
@felisida can I close this one? or is there still something to fix here?

**Comment by felisida** [comment]
yes you can close I wrote in my last post . thanks
![Schermata 2020-11-09 alle 19 14 46](https://user-images.githubusercontent.com/51525150/98580147-d609da80-22bf-11eb-9f4a-4fc64a2126db.jpg)
